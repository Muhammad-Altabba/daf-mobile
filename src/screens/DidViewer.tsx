import * as React from 'react'
import {
  Screen,
  Container,
  Text,
  Button,
  Constants,
  Section,
} from '@kancha/kancha-ui'
import { Mutation } from 'react-apollo'
import { withNavigation, NavigationScreenProps } from 'react-navigation'
import { deleteSeedMutation } from '../lib/Signer'

interface DidViewerProps extends NavigationScreenProps {}

const DidViewer: React.FC<DidViewerProps> = props => {
  // tslint:disable-next-line:no-console
  console.log(props)

  const { navigation } = props
  const did = navigation.getParam('did', 'Did does not exist anymore')
  const address = navigation.getParam(
    'address',
    'Address does not exist anymore',
  )
  const seed = navigation.getParam('seed', 'Seed does not exist anymore')

  return (
    <Screen
      scrollEnabled
      background={Constants.BrandOptions.Secondary}
      safeArea
      footerComponent={
        <Container padding>
          <Mutation mutation={deleteSeedMutation} refetchQueries={['getDids']}>
            {(mutate: any) => (
              <Button
                fullWidth
                type={Constants.BrandOptions.Warning}
                block={Constants.ButtonBlocks.Outlined}
                buttonText={'Delete Seed'}
                onPress={() => {
                  mutate({
                    variables: {
                      address,
                    },
                  }).then(() => props.navigation.goBack())
                }}
                navButton
              />
            )}
          </Mutation>
        </Container>
      }
    >
      <Container>
        <Section title={'DID'}>
          <Container padding>
            <Text>{did}</Text>
          </Container>
        </Section>
        <Section title={'Address'}>
          <Container padding>
            <Text>{address}</Text>
          </Container>
        </Section>
        <Section title={'Seed'}>
          <Container padding>
            <Text>{seed}</Text>
          </Container>
        </Section>
      </Container>
    </Screen>
  )
}

export default withNavigation(DidViewer)
