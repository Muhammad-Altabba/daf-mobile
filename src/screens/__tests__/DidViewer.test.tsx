import 'react-native'
import React from 'react'
import { DidViewer } from '../DidViewer'
import { MockedProvider } from '@apollo/react-testing'
import { render } from 'react-native-testing-library'

jest.useFakeTimers()

const navigation = {
  push: jest.fn(),
  getParam: jest.fn(),
}

it('renders correctly', () => {
  const tree = render(
    <MockedProvider mocks={[]}>
      {
        // @ts-ignore
        <DidViewer navigation={navigation} />
      }
    </MockedProvider>,
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
