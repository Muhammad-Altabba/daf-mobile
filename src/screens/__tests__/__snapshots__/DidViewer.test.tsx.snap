// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />,
  <RCTScrollView
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    contentInsetAdjustmentBehavior="never"
    getTextInputRefs={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onScroll={[Function]}
    scrollEventThrottle={200}
    scrollToBottomOnKBShow={false}
    scrollToInputAdditionalOffset={75}
    startScrolledToBottom={false}
    style={
      Object {
        "backgroundColor": "#EFEEF3",
      }
    }
  >
    <View>
      <View
        pointerEvents="auto"
        style={
          Object {
            "alignItems": undefined,
            "backgroundColor": "#EFEEF3",
            "borderRadius": undefined,
            "flex": 1,
            "flexDirection": undefined,
            "height": undefined,
            "justifyContent": undefined,
            "margin": undefined,
            "marginBottom": undefined,
            "marginLeft": undefined,
            "marginRight": undefined,
            "marginTop": undefined,
            "padding": undefined,
            "paddingBottom": undefined,
            "paddingHorizontal": undefined,
            "paddingLeft": undefined,
            "paddingRight": undefined,
            "paddingTop": undefined,
            "width": undefined,
          }
        }
      >
        <View
          pointerEvents="auto"
          style={
            Object {
              "alignItems": undefined,
              "backgroundColor": undefined,
              "borderRadius": undefined,
              "flex": undefined,
              "flexDirection": undefined,
              "height": undefined,
              "justifyContent": undefined,
              "margin": undefined,
              "marginBottom": undefined,
              "marginLeft": undefined,
              "marginRight": undefined,
              "marginTop": undefined,
              "padding": undefined,
              "paddingBottom": undefined,
              "paddingHorizontal": undefined,
              "paddingLeft": undefined,
              "paddingRight": undefined,
              "paddingTop": undefined,
              "width": undefined,
            }
          }
        >
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": 20,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": undefined,
                  "borderRadius": undefined,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "margin": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": 15,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#BBBBBB",
                    "fontSize": 14,
                    "paddingBottom": 15,
                  }
                }
              >
                DID
              </Text>
            </View>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": "#FFFFFF",
                  "borderBottomColor": "#CCCCCC",
                  "borderBottomWidth": 0.5,
                  "borderRadius": undefined,
                  "borderTopColor": "#CCCCCC",
                  "borderTopWidth": 0.5,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "margin": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": undefined,
                    "borderRadius": undefined,
                    "flex": undefined,
                    "flexDirection": undefined,
                    "height": undefined,
                    "justifyContent": undefined,
                    "margin": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": 15,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "width": undefined,
                  }
                }
              >
                <Text
                  style={Object {}}
                />
              </View>
            </View>
          </View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": 20,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": undefined,
                  "borderRadius": undefined,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "margin": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": 15,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#BBBBBB",
                    "fontSize": 14,
                    "paddingBottom": 15,
                  }
                }
              >
                Address
              </Text>
            </View>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": "#FFFFFF",
                  "borderBottomColor": "#CCCCCC",
                  "borderBottomWidth": 0.5,
                  "borderRadius": undefined,
                  "borderTopColor": "#CCCCCC",
                  "borderTopWidth": 0.5,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "margin": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": undefined,
                    "borderRadius": undefined,
                    "flex": undefined,
                    "flexDirection": undefined,
                    "height": undefined,
                    "justifyContent": undefined,
                    "margin": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": 15,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "width": undefined,
                  }
                }
              >
                <Text
                  style={Object {}}
                />
              </View>
            </View>
          </View>
          <View
            pointerEvents="auto"
            style={
              Object {
                "alignItems": undefined,
                "backgroundColor": undefined,
                "borderRadius": undefined,
                "flex": undefined,
                "flexDirection": undefined,
                "height": undefined,
                "justifyContent": undefined,
                "margin": undefined,
                "marginBottom": undefined,
                "marginLeft": undefined,
                "marginRight": undefined,
                "marginTop": 20,
                "padding": undefined,
                "paddingBottom": undefined,
                "paddingHorizontal": undefined,
                "paddingLeft": undefined,
                "paddingRight": undefined,
                "paddingTop": undefined,
                "width": undefined,
              }
            }
          >
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": undefined,
                  "borderRadius": undefined,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "margin": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": 15,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#BBBBBB",
                    "fontSize": 14,
                    "paddingBottom": 15,
                  }
                }
              >
                Seed
              </Text>
            </View>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "alignItems": undefined,
                  "backgroundColor": "#FFFFFF",
                  "borderBottomColor": "#CCCCCC",
                  "borderBottomWidth": 0.5,
                  "borderRadius": undefined,
                  "borderTopColor": "#CCCCCC",
                  "borderTopWidth": 0.5,
                  "flex": undefined,
                  "flexDirection": undefined,
                  "height": undefined,
                  "justifyContent": undefined,
                  "margin": undefined,
                  "marginBottom": undefined,
                  "marginLeft": undefined,
                  "marginRight": undefined,
                  "marginTop": undefined,
                  "padding": undefined,
                  "paddingBottom": undefined,
                  "paddingHorizontal": undefined,
                  "paddingLeft": undefined,
                  "paddingRight": undefined,
                  "paddingTop": undefined,
                  "width": undefined,
                }
              }
            >
              <View
                pointerEvents="auto"
                style={
                  Object {
                    "alignItems": undefined,
                    "backgroundColor": undefined,
                    "borderRadius": undefined,
                    "flex": undefined,
                    "flexDirection": undefined,
                    "height": undefined,
                    "justifyContent": undefined,
                    "margin": undefined,
                    "marginBottom": undefined,
                    "marginLeft": undefined,
                    "marginRight": undefined,
                    "marginTop": undefined,
                    "padding": 15,
                    "paddingBottom": undefined,
                    "paddingHorizontal": undefined,
                    "paddingLeft": undefined,
                    "paddingRight": undefined,
                    "paddingTop": undefined,
                    "width": undefined,
                  }
                }
              >
                <Text
                  style={Object {}}
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": "#FFFFFF",
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": 15,
        "width": undefined,
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        Object {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "borderRadius": undefined,
          "flex": undefined,
          "flexDirection": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "margin": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "padding": 15,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "width": undefined,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderColor": "#3DCF8A",
            "borderRadius": 8,
            "borderWidth": 2,
            "opacity": 1,
            "padding": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#3DCF8A",
              "fontSize": 20,
            }
          }
        >
          Delete Seed
        </Text>
      </View>
    </View>
  </View>,
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  />,
]
`;
